### PR TITLE
Found avg

### PR DESCRIPTION
**Pull request by MarissaShieh**
@RamKhalsa, would you mind code reviewing this?

It's the barebones React code which renders the average rating of a listing, a search bar, and the container of average criteria ratings. (Note: The test suite is still pretty weak. I may implement those later in a separate pull request.)

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -17,16 +34,43 @@ class App extends React.Component {
...
+      ratings,
+      foundAverage: true,
+    });
+  }
+  
+  filterSearch(searchTerm, filtReviews) {
+    filtReviews = filtReviews || this.state.allReviews;
+    this.setState({
+      reviews: filtReviews,
+      searchTerm: searchTerm,
```
Comment (RamKhalsa) [inline client/components/App.jsx:57]: I don't think the colon + assignment is necessary here!

File: client/components/App.jsx [hunk 2]
```diff
@@ -17,16 +34,43 @@ class App extends React.Component {
...
+    let underSearch;
+    if (this.state.searchTerm !== '') {
+      underSearch = <SearchStats reviews={this.state.reviews} searchTerm={this.state.searchTerm} filterSearch={this.filterSearch} />
+    } else {
+      underSearch = <Ratings ratings={this.state.ratings} foundAverage={this.state.foundAverage} reviews={this.state.reviews} calculateAvg={this.calculateAvg} />
+    }
+
     return (
       <div>
+        <TotalReviews stars={this.state.avgRating} reviews={this.state.reviews}/>
```
Comment (RamKhalsa) [inline client/components/App.jsx:71]: I believe there should be a space after reviews, but before the closing tags!

File: client/components/App.test.jsx [hunk 3]
```diff
@@ -9,20 +9,20 @@ describe('App component rendering', () => {
...
+    let wrapper = shallow(<App />);
+    expect(wrapper.find('TotalReviews').exists()).toBe(true);
+  });
 
-  // it('should contain a Search component', () => {
-  //   let wrapper = shallow(<App />);
-  //   expect(wrapper.find('Search').exists()).toBe(true);
-  // });
+  it('should contain a Search component', () => {
+    let wrapper = shallow(<App />);
```
Comment (RamKhalsa) [inline client/components/App.test.jsx:18]: As a consideration, if you have these under one "describe", you'll only need to set the wrapper once, at the top!

File: client/components/Rating.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+function Rating(props) {
```
Comment (RamKhalsa) [inline client/components/Rating.jsx:3]: What was the trade-off/benefit here of a function declaration vs. a variable defined as a function? I've only ever see the latter, and would be curious on any benefits or differences of this approach!

File: client/components/Rating.test.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { shallow, mount, render } from 'enzyme';
+import Rating from './Rating.jsx';
+
+describe('Rating', () => {
+  it('should render w/o errors', () => {
```
Comment (RamKhalsa) [inline client/components/Rating.test.jsx:6]: Minor point--currently, this tests whether it renders, but I'd recommend adding test cases to test against specific errors for that label. For example, if it renders without the error of rendering undefined as text, I'd push in an undefined prop type and ensure it doesn't show up as text, etc.

File: client/components/Ratings.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Rating from './Rating.jsx';
+
+function Ratings(props) {
+  // if the averages have NOT been calculated before && we have data from DB, then:
+  if(!props.foundAverage && props.reviews.length > 0) {  
+    const numReviews = props.reviews.length;
```
Comment (RamKhalsa) [inline client/components/Ratings.jsx:7]: As a note, you could also do this as: let accuracy, communication, cleanliness, location, checkIn, value;
accuracy = communication = cleanliness = location= checkIn = value = 0!

File: client/components/Search.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    super(props);
+    this.state = {
+      searchTerm: '',
+    };
+  }
+
+  search(e) {
+    e.preventDefault();
+    let searchTerm = this.state.searchTerm;
+    const fitsFilter = this.props.reviews.filter((review) => {
```
Comment (RamKhalsa) [inline client/components/Search.jsx:14]: What were your trade-offs/thoughts on having search locally scoped to this component vs. at the app-level? It seems like this conveniently keeps the Search Term out of App-level state, at the cost of not being able to filter the videos at this level. I'd also be curious if the only reason you need to pass reviews down is to perform line 14. If so, I'd consider bringing that state back to the App-level

**Review by RamKhalsa — APPROVED** [review]
@MarissaShieh -- looks great! I left a few comments, mostly as a mix of curiosity in the trade-offs of a few decisions, and a few less substantive, stylistic comments!

File: client/components/App.test.jsx (hunk 3 above)
Comment (MarissaShieh) [reply client/components/App.test.jsx:18]: I didn't realize this. Thanks!

File: client/components/Rating.jsx (hunk 4 above)
Comment (MarissaShieh) [reply client/components/Rating.jsx:3]: I'm not sure that there is any difference. I know that functions are immediately hoisted, but var/let/const declarations are not. I don't think it affects how the code runs, though.

File: client/components/Rating.test.jsx (hunk 5 above)
Comment (MarissaShieh) [reply client/components/Rating.test.jsx:6]: Will do!

File: client/components/Ratings.jsx (hunk 6 above)
Comment (MarissaShieh) [reply client/components/Ratings.jsx:7]: I didn't realize this. It'll make more code more readable and concise to do this, for sure.

**Comment by MarissaShieh** [comment]
Thanks @RamKhalsa!

**Pull request merged**